lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2087,6 +2089,96 @@ int prArrayContainsSeqColl(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int sc_arrayMaxDepth(PyrSlot *a, int depth)
+{
+	PyrSlot *slot, *endptr;
+	PyrObject *obj;
+	int size, newdepth;
+	newdepth = depth;
+	obj = slotRawObject(a);
+	size = obj->size;
+	slot = obj->slots - 1;
+	endptr = slot + size;
+	while (slot < endptr) {
+		++slot;
+		if (IsObj(slot)) {
+			if (isKindOf(slotRawObject(slot), class_collection)) {
+				if (isKindOf(slotRawObject(slot), class_arrayed_collection)) {
+					newdepth = sc_arrayMaxDepth(slot, depth + 1);
+					if(newdepth < 0) return -1;
+					newdepth = sc_max(depth, newdepth);
+				} else {
+					return -1;  // bail out
+				}
+			}
+		}
+	}
+	return newdepth;
+
+}
+
+int prArrayMaxDepth(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a, *b;
+
+	a = g->sp - 1;
+	b = g->sp;
+	
+	int maxdepth;
+	int err = slotIntVal(b, &maxdepth);
+	if (err) return err;
+	maxdepth = sc_arrayMaxDepth(a, maxdepth);
+	if(maxdepth < 0) return errFailed;
```

Comment:
Same thing with `return`s as above. Also, could there be a more informative error code than `failed` to use here?